src/character/mod.rs

```diff
@@ -152,6 +169,41 @@ impl Character {
             damage / (1 + self.level - receiver.level)
         }
     }
+
+    pub fn receive_status_effect(&mut self, status_effect: StatusEffect) {
+        self.status_effect = status_effect;
+    }
+
+    pub fn maybe_remove_status_effect(&mut self) -> bool {
+        if !self.status_effect.is_normal() {
+            self.receive_status_effect(StatusEffect::Normal);
+            return true;
+        }
+
+        false
+    }
+
+    pub fn maybe_receive_status_effect(&mut self) -> bool {
+        if !self.is_dead() && self.status_effect.is_normal() {
+            let status_effect = random().status_effect();
+            if !status_effect.is_normal() {
+                self.receive_status_effect(status_effect);
+                return true;
+            }
+        }
+
+        false
+    }
+
+    pub fn apply_status_effect(&mut self) -> Attack {
+        match self.status_effect {
+            StatusEffect::Burned(damage) | StatusEffect::Poisoned(damage) => {
+                self.receive_damage(damage);
+                Attack::Effect(self.status_effect)
+            }
+            StatusEffect::Normal | StatusEffect::Confused => Attack::Miss,
```

Comment:
shouldn't `StatusEffect::Normal` produce `Attack::Regular` ? (I get that the downstream logic may expect a miss here, but that's surprising when reading this piece of code)

Reply:
@facundoolano yes it's ugly. Also `StatusEffect::Confused` should be able to be applied, but not as an attack. I'll dig into moving StatusEffect's "apply" logic to a separate thing, as we have with items

Reply:
In case it helps, I would reason about it like this: the Attack enum shouldn't be necessary here since it's more like a battle specific concept, that suggests me that maybe there should be something that maps statis effect -> attack type over in that module. And this side of applying a status effect should deal purely with the taking damage part